src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,688 @@
+            var referencedSymbols = progress.GetReferencedSymbols();
+            var referencedLocations = referencedSymbols.SelectMany(referencedSymbol => referencedSymbol.Locations).Distinct().ToImmutableArray()
+                .OrderByDescending(reference => reference.Location.SourceSpan.Start);
+
+            var list = new List<TInvocationExpressionSyntax>();
+            methodCallSites.Add(document.Document, list);
+            foreach (var refLocation in referencedLocations)
+            {
+                if (!methodCallSites.TryGetValue(refLocation.Document, out var invocations))
+                {
+                    invocations = new List<TInvocationExpressionSyntax>();
+                    methodCallSites.Add(refLocation.Document, invocations);
+                }
+
+                invocations.Add((TInvocationExpressionSyntax)refLocation.Location.FindNode(cancellationToken).GetRequiredParent());
+            }
+
+            return methodCallSites.ToImmutableDictionary();
+        }
+
+        private static SeparatedSyntaxList<SyntaxNode> AddArgumentToArgumentList(
+            SeparatedSyntaxList<SyntaxNode> invocationArguments, SyntaxGenerator generator,
+            SyntaxNode newArgumentExpression, int insertionIndex, string name, bool named)
+        {
+            SyntaxNode argument;
+            if (named)
+            {
+                argument = generator.Argument(name, RefKind.None, newArgumentExpression.WithAdditionalAnnotations(Simplifier.Annotation));
+            }
+            else
+            {
+                argument = generator.Argument(newArgumentExpression.WithAdditionalAnnotations(Simplifier.Annotation));
+            }
+
+            return invocationArguments.Insert(insertionIndex, argument);
+        }
+
+        private static ImmutableArray<SyntaxNode> AddExpressionArgumentToArgumentList(
+            ImmutableArray<SyntaxNode> arguments, SyntaxNode expression, SyntaxGenerator generator)
+        {
+            var newArgument = generator.Argument(expression);
+            return arguments.Add(newArgument);
+        }
+
+        /// <summary>
+        /// Generates a method declaration containing a return expression of the highlighted expression
+        /// </summary>
+        private void GenerateNewMethod(SemanticDocument semanticDocument, TExpressionSyntax expression,
+            IMethodSymbol methodSymbol, string newMethodIdentifier, SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var document = semanticDocument.Document;
+            var syntaxFacts = semanticDocument.Document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var oldMethodDeclaration = expression.FirstAncestorOrSelf<SyntaxNode>(node => IsContainedInParameterizedDeclaration(node), true)!;
+            var returnExpression = new List<SyntaxNode>
+            {
+                editor.Generator.ReturnStatement(expression.WithoutTrailingTrivia())
```

Comment:
doc why you remove trailing trivia (but not leading trivia for example).

Reply:
Added